src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -190,7 +190,6 @@
     protected void finalize() throws Throwable {
         if (executor != null && !executor.isShutdown()) {
```

Comment:
`finalize()` method is called even though constructor fails. the executor is instantiated in the constructor.